celery/app/base.py

```diff
@@ -836,7 +836,13 @@ def send_task(self, name, args=None, kwargs=None, countdown=None,
         """
         parent = have_parent = None
         amqp = self.amqp
-        task_id = task_id or uuid()
+        if task_id is None:
```

Comment:
We need to add tests to make sure it works with all [canvas primitives](https://docs.celeryq.dev/en/stable/userguide/canvas.html#the-primitives).
E.g, setting `task_id_generator` once and ensuring all primitives are using it, and another test with a canvas workflow, ensuring as well the generator is used as expected.

I’m not sure this (`send_task`) is enough to cover all cases